packages/rest-crud/src/crud-rest-builder.plugin.ts

```diff
@@ -0,0 +1,118 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/rest-crud
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {bind, ControllerClass, inject} from '@loopback/core';
+import {
+  asModelApiBuilder,
+  ModelApiBuilder,
+  ModelApiConfig,
+} from '@loopback/model-api-builder';
+import {
+  ApplicationWithRepositories,
+  Class,
+  Entity,
+  EntityCrudRepository,
+} from '@loopback/repository';
+import {Model} from '@loopback/rest';
+import debugFactory from 'debug';
+import {defineCrudRepositoryClass, defineCrudRestController} from '.';
+
+const debug = debugFactory('loopback:boot:crud-rest');
+
+export interface ModelCrudRestApiConfig extends ModelApiConfig {
+  // E.g. '/products'
+  basePath: string;
+}
+
+@bind(asModelApiBuilder)
+export class CrudRestApiBuilder implements ModelApiBuilder {
+  readonly pattern: string = 'CrudRest';
+
+  build(
+    application: ApplicationWithRepositories,
+    modelClass: typeof Model & {prototype: Model},
+    cfg: ModelApiConfig,
+  ): Promise<void> {
+    const modelName = modelClass.name;
+    const config = cfg as ModelCrudRestApiConfig;
+    if (!config.basePath) {
+      throw new Error(
+        `Missing required field "basePath" in configuration for model ${modelName}.`,
+      );
+    }
+
+    if (!(modelClass.prototype instanceof Entity)) {
+      throw new Error(
+        `CrudRestController requires a model that extends 'Entity'. (Model name ${modelName} does not extend 'Entity')`,
+      );
+    }
+    const entityClass = modelClass as typeof Entity & {prototype: Entity};
+
+    let repoClassName = entityClass.name + 'Repository';
+
+    if (application.isBound('repositories.' + repoClassName)) {
```

Comment:
Please add a test to verify this scenario. AFAICT, it's not covered now.

https://coveralls.io/builds/28705524/source?filename=packages/rest-crud/src/crud-rest-builder.plugin.ts#L56

Reply:
I couldn't find a good way to test this (https://github.com/strongloop/loopback-next/pull/4589#discussion_r378684011) since the binding for both repositories would be the same since they would share the same name so I don't know if there's a way to verify it's using the repository that was bound before. I have an attempt in [this commit](https://github.com/strongloop/loopback-next/pull/4589/commits/324606155a1ae8ee8e8f1160a1782dbdffa41ec0#diff-858d1af4393d324a034e5a75fbdd5b64R60).